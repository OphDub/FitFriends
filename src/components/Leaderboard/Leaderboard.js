import React from 'react';
import './Leaderboard.css';
import { Teammate } from '../Teammate/Teammate';

export const Leaderboard = ({topThree}) => {
  const renderedLeaders = topThree.map(leader => 
    <Teammate {...leader}/>
  )

  return (
    <section className="Leaderboard">
      <h3>Leaderboard</h3>
      {renderedLeaders}
    </section>
  )
};