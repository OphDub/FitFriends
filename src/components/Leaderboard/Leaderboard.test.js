import React from 'react';
import { shallow } from 'enzyme';
import { Leaderboard } from './Leaderboard';

describe('LEADERBOARD', () => {
  it('should match snapshot', () => {
    const renderedComponent = shallow(<Leaderboard />);

    expect(renderedComponent).toMatchSnapshot();
  });
});