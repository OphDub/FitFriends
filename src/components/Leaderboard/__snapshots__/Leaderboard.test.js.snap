// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LEADERBOARD should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    topThree={
        Array [
            Object {
              "UID": "randomKey1",
              "name": "Felipe",
              "workoutTally": 6,
              "workoutsPosted": 2,
            },
            Object {
              "UID": "randomKey2",
              "name": "Fion",
              "workoutTally": 5,
              "workoutsPosted": 3,
            },
            Object {
              "UID": "randomKey3",
              "name": "Morgan",
              "workoutTally": 3,
              "workoutsPosted": 4,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Leaderboard
</h3>,
        Array [
          <Unknown
            UID="randomKey1"
            name="Felipe"
            workoutTally={6}
            workoutsPosted={2}
/>,
          <Unknown
            UID="randomKey2"
            name="Fion"
            workoutTally={5}
            workoutsPosted={3}
/>,
          <Unknown
            UID="randomKey3"
            name="Morgan"
            workoutTally={3}
            workoutsPosted={4}
/>,
        ],
      ],
      "className": "Leaderboard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Leaderboard",
        },
        "ref": null,
        "rendered": "Leaderboard",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": "randomKey1",
        "nodeType": "function",
        "props": Object {
          "UID": "randomKey1",
          "name": "Felipe",
          "workoutTally": 6,
          "workoutsPosted": 2,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "randomKey2",
        "nodeType": "function",
        "props": Object {
          "UID": "randomKey2",
          "name": "Fion",
          "workoutTally": 5,
          "workoutsPosted": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "randomKey3",
        "nodeType": "function",
        "props": Object {
          "UID": "randomKey3",
          "name": "Morgan",
          "workoutTally": 3,
          "workoutsPosted": 4,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Leaderboard
</h3>,
          Array [
            <Unknown
              UID="randomKey1"
              name="Felipe"
              workoutTally={6}
              workoutsPosted={2}
/>,
            <Unknown
              UID="randomKey2"
              name="Fion"
              workoutTally={5}
              workoutsPosted={3}
/>,
            <Unknown
              UID="randomKey3"
              name="Morgan"
              workoutTally={3}
              workoutsPosted={4}
/>,
          ],
        ],
        "className": "Leaderboard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Leaderboard",
          },
          "ref": null,
          "rendered": "Leaderboard",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": "randomKey1",
          "nodeType": "function",
          "props": Object {
            "UID": "randomKey1",
            "name": "Felipe",
            "workoutTally": 6,
            "workoutsPosted": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "randomKey2",
          "nodeType": "function",
          "props": Object {
            "UID": "randomKey2",
            "name": "Fion",
            "workoutTally": 5,
            "workoutsPosted": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "randomKey3",
          "nodeType": "function",
          "props": Object {
            "UID": "randomKey3",
            "name": "Morgan",
            "workoutTally": 3,
            "workoutsPosted": 4,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
